automation/orchestrator.go

```diff
@@ -196,13 +200,28 @@ func (o *Orchestrator) Shutdown() {
 // represented as a string
 func (o *Orchestrator) handleTrigger(ctx context.Context, e event.Event) error {
 	if e.Type == event.ETWorkflowTrigger {
-		wid, ok := e.Payload.(string)
+		wtp, ok := e.Payload.(*event.WorkflowTriggerPayload)
 		if !ok {
-			return fmt.Errorf("handleTrigger: expected event.Payload to be a string: %v", e.Payload)
+			return fmt.Errorf("handleTrigger: expected event.Payload to be an `event.WorkflowTriggerPayload`: %v", e.Payload)
 		}
 		go func() {
-			if err := o.RunWorkflow(ctx, workflow.ID(wid)); err != nil {
-				log.Errorf("%s", err)
+			wf, err := o.GetWorkflow(workflow.ID(wtp.WorkflowID))
+			if err != nil {
+				log.Debugw("handleTrigger: error fetching workflow", "id", wtp.WorkflowID, "err", err)
+				return
+			}
+			for _, trig := range wf.Triggers {
+				if trig.ID() == wtp.TriggerID {
+					trig.Advance()
+					break
+				}
+			}
+			wf, err = o.SaveWorkflow(wf)
+			if err != nil {
+				log.Debugw("handleTrigger: error saving workflow", "id", wtp.WorkflowID, "err", err)
+			}
+			if err := o.runWorkflow(ctx, wf); err != nil {
+				log.Debugw("handleTrigger: error running workflow", "err", err)
```

Comment:
I foresee some potential bugs where a user can edit a `Workflow` after it was triggered but before it is run. This worries me less in a 

I think we need to add "locking" workflows for editing to the responsibilities of the future "sync" system.